cluster-autoscaler/cloudprovider/azure/azure_scale_set.go

```diff
@@ -35,8 +36,29 @@ import (
 	schedulernodeinfo "k8s.io/kubernetes/pkg/scheduler/nodeinfo"
 
 	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2018-10-01/compute"
+	"github.com/Azure/go-autorest/autorest"
 )
 
+var (
+	vmssSizeRefreshPeriod      = 15 * time.Second
+	vmssInstancesRefreshPeriod = 5 * time.Minute
+)
+
+func init() {
+	// In go-autorest SDK https://github.com/Azure/go-autorest/blob/master/autorest/sender.go#L242,
+	// if ARM returns http.StatusTooManyRequests, the sender doesn't increase the retry attempt count,
+	// hence the Azure clients will keep retrying forever until it get a status code other than 429.
+	// So we explicitly removes http.StatusTooManyRequests from autorest.StatusCodesForRetry.
+	// Refer https://github.com/Azure/go-autorest/issues/398.
+	statusCodesForRetry := make([]int, 0)
+	for _, code := range autorest.StatusCodesForRetry {
```

Comment:
we should update the retry status code for azure cloud provider here:
https://github.com/kubernetes/kubernetes/blob/bf86300241e51f011040d208e43f1cd2a5c0378a/staging/src/k8s.io/legacy-cloud-providers/azure/azure_backoff.go#L586-L589

Reply:
The above retry logic is ok, since there's already a backoff-retry there. And for 412, cloud provider should retry actually, just with backoff delay.